ios/RNDeviceTimeFormat.m

```diff
@@ -14,11 +14,16 @@ - (dispatch_queue_t)methodQueue
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
 {
-    NSString *formatStringForHours = [NSDateFormatter dateFormatFromTemplate:@"j" options:0 locale:[NSLocale currentLocale]];
-    NSRange containsA = [formatStringForHours rangeOfString:@"a"];
-    BOOL is24HourFormat = (containsA.location == NSNotFound);
-
-    resolve(@(is24HourFormat));
+    @try {
+        NSString *formatStringForHours = [NSDateFormatter dateFormatFromTemplate:@"j" options:0 locale:[NSLocale currentLocale]];
+        NSRange containsA = [formatStringForHours rangeOfString:@"a"];
+        BOOL is24HourFormat = (containsA.location == NSNotFound);
+        
+        resolve(@(is24HourFormat));
+    }
+    @catch (NSException *exception) {
+        reject(@"exception", nil, RCTErrorWithMessage(@(exception.reason));
+    }
 }
```

Comment:
@casperstorm does the errorhandling look ok? or is the @try/@catch complete overkill?

Reply:
I think its good, it's not overkill but rarely seen in Obj-C. Let's go with it.